### PR TITLE
Skip initial part of replay

### PR DESCRIPTION
**Pull request by krisgry**
Added functionality to skip the initial <skip> seconds of a log replay, inspired by dune-lsfreplay. This is done by simply looping through the messages in the log file, until the difference between timestamp of the message and the start of the log is larger than the desired number of seconds to skip. While looping through messages, EntityInfo messages are handled, to build a map of the Entities.

This has been tested, and works for the functionality that I needed, but perhaps there is something I've missed?

### CODE REVIEW

File: src/Transports/Replay/Task.cpp [hunk 1]
```diff
@@ -47,6 +47,7 @@ namespace Transports
       std::string startup_file;
       std::vector<std::string> msgs;
       std::vector<std::string> ents;
+      double skip;
```
Comment (krisklau) [inline src/Transports/Replay/Task.cpp]: A bit undescriptive, how about `initial_log_skip_seconds`?

File: src/Transports/Replay/Task.cpp [hunk 2]
```diff
@@ -261,8 +267,39 @@ namespace Transports
...
+          if (!m)
+          {
+            err("No messages for specified time range");
+            return;
+          }
+          else
+            inf("Skipped messages up to %s", Time::Format::getTimeDate(m->getTimeStamp()).c_str());
+        }
+
+        //
```
Comment (krisklau) [inline src/Transports/Replay/Task.cpp]: Remove

File: src/Transports/Replay/Task.cpp [hunk 3]
```diff
@@ -407,15 +421,7 @@ namespace Transports
...
-              IMC::EntityInfo* ei = static_cast<IMC::EntityInfo*>(m);
-              Name2Eid::iterator itr = m_name2eid.find(ei->label);
-
-              if (itr != m_name2eid.end())
-              {
-                m_eid2eid[ei->id] = itr->second;
-
-                trace("entity %s %d --> %d", ei->label.c_str(), (int)ei->id, (int)itr->second);
-              }
+              updateEntityMap(m);
```
Comment (krisklau) [inline src/Transports/Replay/Task.cpp]: Consider moving the method implementation after the usage for readability.

**Comment by krisgry** [comment]
Fixed according to your comments, @krisklau. Thanks 👍

**Comment by tsmarques** [comment]
Hi,

à priori everything seems ok. Juts two comments:

- We sometimes fail this ourselves, but when committing we try to start with a capital word after **:** and end with a **.**; No need to change now.

- The method `startReplay` is getting a bit big so I'd just move this new feature into its own method.

**Comment by luis109** [comment]
Hi,
I just tested your PR and it seems fine. I only ask to:

-  Move the "skip messages" in `startReplay` into its own method, as stated in the previous comment.

- Add default configuration (Seconds to skip = 0.0) to etc/testing/replays/sgnav-replay.ini.

PS: Just a heads up you might find useful, you can now speed up replay (and simulation) since PR #193 merge.

**Comment by krisgry** [comment]
Finally got around to fix this :) 

Thanks for the feedback @luis109 and @tsmarques

**Pull request merged**